packages/uicore/src/components/TagInput/TagInput.tsx

```diff
@@ -284,7 +274,7 @@ export function TagInput<T>(props: TagInputProps<T>) {
       createNewItemFromQuery={(allowNewTag && itemFromNewTag) || undefined}
       createNewItemRenderer={renderCreateNewTag}
       onItemSelect={onItemSelect}
-      tagRenderer={(item: T) => labelFor(getLabelFromItems(item, items))}
```

Comment:
`getLabelFromItems` loops and finds the matching item, but `tagRenderer` allows accessing the correct item via the argument, making `getLabelFromItems` unnecessary.